src/coreclr/jit/target.h

```diff
@@ -612,6 +611,7 @@ inline regMaskTP genRegMask(regNumber regNum, var_types type)
     }
     else
     {
+        assert(varTypeUsesIntReg(type));
```

Comment:
Should this also be reverseed to have `varTypeUsesIntReg()` check first? Also, shouldn't this include a case for kmask register too?

Reply:
This is for `TARGET_ARM` (note the condition is `#ifndef`). We could inverse it, but since its ARM only I didn't see the need.

Reply:
Inverse and made it more obvious that path is for TARGET_ARM